src/Faker/Provider/id_ID/Person.php

```diff
@@ -253,4 +253,41 @@ public static function suffix()
     {
         return static::randomElement(static::$suffix);
     }
+
+    /**
+     * Generates Nomor Induk Kependudukan (NIK)
+     *
+     * @link https://en.wikipedia.org/wiki/National_identification_number#Indonesia
+     *
+     * @param null|string $gender
+     * @param null|\DateTime $birthDate
+     * @return string
+     */
+    public function nik($gender = null, $birthDate = null)
+    {
+        # generate first numbers (region data)
+        $nik = $this->generator->numerify('######');
+
+        if (!$birthDate) {
+            $birthDate = $this->generator->dateTimeBetween();
+        }
+
+        if (!$gender) {
+            $gender = $this->generator->randomElement([self::GENDER_MALE, self::GENDER_FEMALE]);
```

Comment:
Faker keeps compatibility with PHP 5.3. Please use `Array()` instead of `[]` (this is why the unit tests fail).